### PR TITLE
Add OpInfos for combinations, cartesian_prod, sum_to_size, ldexp, as_strided

### PR DESCRIPTION
**Pull request by saketh-are**

### CODE REVIEW

File: test/test_fx_experimental.py [hunk 1]
```diff
@@ -1463,8 +1463,10 @@ def test_normalize_operator_exhaustive(self, device, dtype, op):
         # Sorted and one entry on each line to minimize merge conflicts.
         op_skip = {
```
Comment (mruberry) [inline test/test_fx_experimental.py:1464]: cc @Chillee We should kill this list and use regular DecorateInfo skips if possible

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -1411,6 +1411,73 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
                             dim=(0, 1))))
         return inputs
 
+def sample_inputs_as_strided(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(torch.tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # input shape, output shape, output stride, output storage offset
+    test_cases = [
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1418]: This is cool but maybe a few more samples, like the examples in the function's documentation, would be interesting, too?

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -1411,6 +1411,73 @@ def sample_inputs_linalg_norm(op_info, device, dtype, requires_grad):
...
+        kwargs = dict(r=r, with_replacement=with_replacement)
+
+        samples.append(SampleInput(input_t, kwargs=kwargs))
+
+    return tuple(samples)
+
+def sample_inputs_cartesian_prod(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(torch.tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    # constructs 1-D tensors with varying number of elements
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1455]: Nice comments

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -6533,6 +6600,26 @@ def generator():
...
 
 
+def sample_inputs_sum_to_size(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
+
+    # list of tuples (shape, shape) defining the shapes of the input and output tensors
+    sample_shapes = [
+        ((), ()),
+        ((S), (1)),
+        ((S, S), (1, 1)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:6610]: Nice broadcasting tests

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -6533,6 +6600,26 @@ def generator():
...
+        ((S, S), (1, S)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:6611]: How about this one without broadcasting?

File: torch/testing/_internal/common_methods_invocations.py [hunk 6]
```diff
@@ -6533,6 +6600,26 @@ def generator():
...
+        ((S, S, S), (S, S, 1)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:6612]: Maybe let's put the 1 in the middle for fun

File: torch/testing/_internal/common_methods_invocations.py [hunk 7]
```diff
@@ -8223,6 +8310,24 @@ def ref_pairwise_distance(input1, input2):
...
+           supports_out=False,
+           sample_inputs_func=sample_inputs_combinations),
+    OpInfo('cartesian_prod',
+           op=torch.cartesian_prod,
+           dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           supports_autograd=False,
+           supports_out=False,
+           sample_inputs_func=sample_inputs_cartesian_prod,
+           skips=(
+               # RuntimeError: input->type()->kind() == TypeKind::OptionalType
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:8326]: Nice comment. cc @eellison

File: torch/testing/_internal/common_methods_invocations.py [hunk 8]
```diff
@@ -8290,6 +8395,12 @@ def ref_pairwise_distance(input1, input2):
            autodiff_fusible_nodes=['aten::contiguous'],
            assert_jit_shape_analysis=True,
            supports_out=False),
+    OpInfo('sum_to_size',
+           op=lambda x, *args, **kwargs: x.sum_to_size(*args, **kwargs),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:8399]: Nice function construction

File: torch/testing/_internal/common_methods_invocations.py [hunk 9]
```diff
@@ -9335,6 +9446,29 @@ def ref_pairwise_distance(input1, input2):
...
                                     dtypes=[torch.cfloat, torch.cdouble]),
                    )),
+    BinaryUfuncInfo('ldexp',
+                    dtypes=all_types_and_complex_and(torch.bool, torch.half, torch.bfloat16),
+                    supports_forward_ad=True,
+                    supports_inplace_autograd=False,
+                    sample_inputs_func=sample_inputs_binary_pwise,
+                    supports_out=True,
+                    skips=(
+                        # RuntimeError: mul(): functions with out=... arguments don't support
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:9456]: Interesting... would you file and cite an issue for this? 

Can these be converted to xfails?

File: torch/testing/_internal/common_methods_invocations.py [hunk 10]
```diff
@@ -9820,6 +9954,18 @@ def ref_pairwise_distance(input1, input2):
...
                DecorateInfo(unittest.skip("Skipped!"), 'TestCommon', 'test_out'),
            )),
+    OpInfo('as_strided',
+           op=lambda x, size, stride, storage_offset=0:
+               torch.as_strided(x, size, stride, storage_offset=storage_offset),
+           dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16),
+           supports_out=False,
+           sample_inputs_func=sample_inputs_as_strided,
+           skips=(
+               # FIXME: AssertionError: False is not true : Tensors failed to compare as equal!
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:9964]: cc @anjali411 -- that's weird

**Review by mruberry — COMMENTED** [review]
Overall looks really good (as usual) @saketh-are, just a few inline questions/suggestions and ccs for some of our colleagues.

cc @krshrimali who I know was looking at OpInfoing some of these, too (we should start marking stuff in the spreadsheet better!). @krshrimali may want to review, too

**Review by t10-13rocket — APPROVED** [review]
LGTM

**Comment by saketh-are** [comment]
Failing check is related to `linalg_lu_factor` changes which were reverted in master

**Review by mruberry — APPROVED** [review]
Cool! Might be worth filing a separate issue for the test_neg_view failure of as_strided -- is that a silent correctness issue? I could believe that test_noncontiguous_samples fails because of how it's setting strides.

**Review by krshrimali — APPROVED** [review]

**Pull request closed without merging**